### ISSUE TITLE
Add Transportation Mitigation Fees codes to AMANDA (NN)

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1729]
### As a TIA Case Manager, I don't want the system to accept an approved attempt on ATD Engineering Review if there are unpaid ATD fees on the folder. This need applies to folder types SP, C8, ZC, and C. 
----

**Stakeholders: Sangeeta, Nazlie**
**DTS Manager: Nate**

**Acceptance Criteria:**
- 'Transportation Mitigation Fee-in-lieu' is available on TIA, Insert & Run fee procedure
- ATD Engineering Review process does not save an 'Approved' attempt result if there are unpaid ATD fees on the fee tab. This applies to folders SP, C8, ZC, and C.
- ATD Engineering Review process provides attempt result 'Insert Mitigation Fee'
- The Insert Mitigation Fee inserts the Transportation Mitigation Fee-in-lieu in the amount specified by the case manager on the process info. 

DTS URL: https://atd.knack.com/dts#service-requests/view-issue-details/5cb4cd404bf74b2b9aa71a3c

*Migrated from [atd-amanda #61](https://github.com/cityofaustin/atd-amanda/issues/61)*

**Comment by atdservicebot** [comment]
*From atdservicebot on 2019-05-10*:

Email Sent to Neil Galati w/ DSD on 4/9:
   -(Similar email also sent to Wendy Rhoades w/ PAZ)

Hello Neil,
 
ATD has implemented a new process that allows customers to pay a fee in lieu for transportation mitigations. The fee will need to be added to SP, C8, ZC, and TIA folders in AMANDA.
 
Neil, I will submit a CTM ticket for development of these fees however we will need DSD approval for adding the fee to SP & C8. Below is more info about the fee; please let me know if DSD has any concerns about this, or if we can move forward with the CTM request. (Approval for the ZC folder will be requested from PAZ.)
 
Fee Information:
- Name: ATD Mitigation Fee in Lieu
- FDU: 7070-2400-9100-4163
- Amount: Manually Calculated by TIA staff

Created By: Tracy Linder
Created Date: 2019-04-15 13:30:00

**Comment by atdservicebot** [comment]
*From atdservicebot on 2019-05-10*:

4/9/19 Notes From 1st Meeting W/ Sangeeta & Nazlee

New Transportation Fee
	Need to go into AMANDA
	The FDU is 7070-2400-9100-4163 
	Only 1 fee going into this: ATD Mitigation Fee in Lieu
	What folder does this apply to?
		SP, TIA, ZC, C8
		Need to get DSD approval to add to SP & C8. Sangeeta suggested Neil Gallati. 
		PZD - Talk to Wendy Rhoades
	Not a flat fee, there is a calculation. 
		It will have to be manually calculated 100% of the time.
	This fee is already being collected
	Estimate: 6 Weeks to Production
	
User Account Updates For Mitigation Fees:
	Fee permissions, update folder, add attachments
	Nazlee Saeedi, Sangeeta Jain, Austin Jones, Nathan Aubert, Upal Barua, Joan Jenkins,
	
Training Needed:
	When fees are ready to go live, the team will need training, including proper invoicing methods


Created By: Tracy Linder
Created Date: 2019-04-15 13:30:00

**Comment by atdservicebot** [comment]
*From atdservicebot on 2019-05-10*:

4/15 Email Responses sent to Neil & Wendy:

Hi Wendy, 

Thank you for your quick response on this. I spoke with Sangeeta and Nazlie to confirm the correct answers. ATD is currently working with Roderick so that all ATD reviews on SP and ZC folders will be assigned to ATD staff for review. A high level overview of the process related to this fee is below. Regarding ZC versus the TIA folder; TIA folders are only created to document an analysis, which is only required in about 20% of SP/ZC reviews. This new fee will need to be collected when an analysis is not being performed, so it makes sense to collect the fee on the folder related to the review being performed.

1.       Folder process is assigned to ATD staff
2.       ATD staff identifies the fee requirement, and
a.       Manually adds the fee to the ZC fee tab, and creates an invoice
b.       Provides the customer with an invoice, with instructions to pay an ATD cashier either at our Barton Oaks or Toomey Rd locations
3.       ATD staff will sign-off (approve) their folder process after the fee has been paid 

If there are any more questions we’d like to schedule a quick call to get them all answered. Thank you again.

--------

Hi Neil,

Answers to your questions are below. If any more questions come up we’d like to schedule a quick call to get it all covered.

Would this fee be added manually by ATD staff using the Add Fee button or should this be inserted automatically?
Fee added manually by ATD staff using Add Fee button
When should this fee get inserted?
ATD TIA staff will insert this fee during their review
When will the ATD staff calculate the amount? (Folder status)
When the folder status is 'In Review'
Where will this amount be stored in AMANDA?
Amount will always be manually calculated - no amount will be stored.
If not automatic, who will insert the fee?
ATD TIA staff
Will a physical invoice get created for in person payments? When and by whom?
ATD TIA staff will invoice the fee after inserting it, provide the customer with an invoice with instructions to pay an ATD cashier either at our Barton Oaks or Toomey Rd locations.
When must the payment get collected?
The applicant must pay the fee before ATD TIA staff will approve the TIA review process assigned to them.

Created By: Tracy Linder
Created Date: 2019-04-15 13:31:00

**Comment by atdservicebot** [comment]
*From atdservicebot on 2019-05-10*:

4/15 Update

Nazlee, Sangeeta, and I spoke to discuss the questions we got from DSD & PAZ. Afterwards I (Tracy) responded to Wendy & Neil's emails. 

Neil's Questions
	• Would this fee be added manually by ATD staff using the Add Fee button or should this be inserted automatically?
		○ Fee added manually by ATD staff using Add Fee button
	• When should this fee get inserted?
		○ ATD TIA staff will insert this fee during their review
	• When will the ATD staff calculate the amount? (Folder status)
		○ When the folder status is 'In Review'
	• Where will this amount be stored in AMANDA?
		○ Amount will always be manually calculated - no amount will be stored.
	• If not automatic, who will insert the fee?
		○ ATD TIA staff
	• Will a physical invoice get created for in person payments? When and by whom?
		○ ATD TIA staff will invoice the fee after inserting it, provide the customer with an invoice with instructions to pay an ATD cashier either at our Barton Oaks or Toomey Rd locations. 
	• When must the payment get collected?
		○ The applicant must pay the fee before ATD TIA staff will approve the TIA review process assigned to them. 

Wendy's Questions:
	• how ATD staff will inform the zoning case manager of the added fee and the timeframe to collect that fee
		
	• The zoning case managers will need an overview from ATD of how this is envisioned to work and which department is responsible for collecting the fee
	• why is this fee to be collected through the ZC folder rather than the related TIA folder 


Created By: Tracy Linder
Created Date: 2019-04-15 13:33:00

**Comment by SurbhiBakshi** [comment]
LIT-TDS Meeting scheduled to gather more details on 04/23.

**Comment by amenity** [comment]
Need to assign new fee codes to Transportation _mitigation_  fees.

**Comment by TracyLinder** [comment]
Meeting rescheduled for 5/1

**Comment by TracyLinder** [comment]
**Meeting Notes**

Tranportation Mitigation fees are collected during TDS review of a project. Fees have been done manually, outside of AMANDA. 

- Sangeeta wants the fees added, coded to the folder, and for the folder attempt 'approved' to validate that all ATD fees on the folder are paid before the attempt is saved. 
- Sangeeta will send Sree the list of fee codes. 
- Sree needs the GL fee codes for the new fees. Sangeeta will send them. 

**Acceptance Criteria:** 
- Fees are available on insert & run menu on various folders
- Fees are payable online 
- An approved attempt on Sangeeta's process does not pay if any ATD fees on the folder are outstanding.

**Comment by TracyLinder** [comment]
**Email to Sangeeta Today**

Hi Sangeeta, 

This is a follow-up question for ticket #1729, add TIA mitigation fee to AMANDA. On the C folder (Subdivision Joint City/County), what is the name of the folder process that you use to document your review? Sree did some research, and it looks like for the C folder, it’s a different process than all the others. 

Thank You,

Tracy Linder, Product Manager

**Comment by TracyLinder** [comment]
Sree still working on this one. TIA update done, 3 other folders left to go.

**Comment by TracyLinder** [comment]
Notes For Sree:
1. Remove these features from the TIA folder completely. 
2. Change fee name from 'TIA Mitigation...' to 'Transportation Mitigation Fee-in-lieu'
3. Make sure this fee is available on the Insert & Run procedure on TIA folder. 
3. Remove the 'calculate fees' and folder info features from Initial Intake. Put it on ATD Engineering only for all folder types other than TIA. 
4. Change the name of the attempt result from 'calculate fees' to 'Insert Mitigation Fee

**Comment by TracyLinder** [comment]
Sent demo last week; received notes from Tracy & team. Will iterate & send out demo during Sprint 26.

**Comment by TracyLinder** [comment]
Today Sree demoed her progress so far for Nate & me. All items from my previous note have been addressed (and it's looking great). Sree will work on applying the new code to ZC, C8, C folders, then we will be ready for stakeholder input.

**Comment by TracyLinder** [comment]
The issue description has been updated with the most current user stories and acceptance criteria.

**Comment by TracyLinder** [comment]
### Demo Meeting 5/28/20
**Info:** A meeting with stakeholders, LaunchIT, and DTS to review current prototype (DEV) and collect user feedback.**
**Attendees:** Tracy, Nate, Sree, Sangeeta, Nazlie, Kaustubh, Amber

@Nadin-Nader do you have any notes from this meeting that you can add to this issue? I'll look for mine too. Thanks!

**Comment by TracyLinder** [comment]
It doesn't look like the meeting on 5/28 was recorded.

**Comment by TracyLinder** [comment]
@Nadin-Nader disregard the request for notes. I just realized my comment above (Notes for Sree) were my notes from that meeting. 😆

**Comment by TracyLinder** [comment]
### Update

**Sree emailed us a demo of this work after her latest updates on 6/5, requesting approval to promote to Dev.**



**My Response to her demo, via email today:**
Hi Sree,

I am sorry for my delayed response on this, and appreciate your sending me a message about it on Teams. I have reviewed the demo, and my feedback is below (two things).

Can you please run the following test, perhaps on an SP folder, and let me know if this feature still works as expected? (An email is fine, no demo needed.)
Test Steps
1. Add any ATD fee (from the list)
2. Create bill for the fee
3. Cancel the bill
4. Save 'Approved' attempt on ATD Engineering Review

I’m also now wondering if saving the attempt 'Insert Mitigation Fee' should also prompt the system to create a bill for the fee after it is inserted. I’m going to send Sangeeta an email asking for feedback and copy you; Sangeeta usually responds in the morning next day. 

Thank You,

Tracy Linder, Product Manager

**Follow-Up Email to Sree:**

Hi Sree,

Please disregard my last comment about the system creating the bill. If that surfaces as a need during UAT, I’ll create a new issue for it to be completed during a later sprint. 

All the Best,

Tracy Linder, Product Manager

**Comment by TracyLinder** [comment]
** Update**

Sree ran the test I requested, and there were no issues to report. I've asked her to move this to Test.

**Comment by TracyLinder** [comment]
@Nadin-Nader will be helping Sree w/ deployment to Dev. I'm also updating milestone to 27.

**Comment by Nadin-Nader** [comment]
This has been sent to CTM DevOps CM Team to deploy (SCTASK0076106). Waiting

**Comment by Sreelatha-BlazeKannan** [comment]
had a commit conflict with PROCESS_SP_0000005 and reverted and committed the whole procedures again on 06-23-2020

**Comment by Nadin-Nader** [comment]
**Status:**

Service Now TASK for AMANDA CM Team is SCTASK0076819. Waiting for it to be deployed

**Comment by Nadin-Nader** [comment]
Hi Martin,

Seems like I am confused with all of the different commits. I was looking at this and found that we have TWO tasks for this SCTASK0076819 (this one is in your To-do column); and SCTASK0076106 (this one is in your Merge Conflicts column) can you let me know which one you would like to keep open for these commits and can you tell me what the commits are?

From: Aponte, Martin <Martin.Aponte@austintexas.gov> 
Sent: Thursday, June 25, 2020 9:10 AM
To: Sreelatha.b@launchitcorp.com; Nader, Nadin <Nadin.Nader@austintexas.gov>
Subject: RE: Committed Again as two commits 

Hello
Commits have been promoted and on dev

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder  @Nadin-Nader - #1729 – Transportation Mitigation Fee-in-Lieu successfully moved to DEV and tested all components in DEV environment as well. Please TEST in DEV and let us know if this can be moved to TEST Environment.

**Comment by TracyLinder** [comment]
**Status:** Currently in DEV pending Tracy review/approval to promote.

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder  - Please do let us know whether it is ready to move from DEV to TEST

**Comment by TracyLinder** [comment]
@Sreelatha-BlazeKannan  @Nadin-Nader I don't have the DEV permissions I need to test/approve this. It won't even let me change my password. Nate this is #3206 (update my DEV permissions.)

**Comment by Nadin-Nader** [comment]
@TracyLinder  your permissions have been updated on DEV :)

**Comment by TracyLinder** [comment]
### This has been approved to promote to Prod. 🎉

**Comment by TracyLinder** [comment]
Martin w/ CTM informed @Nadin-Nader that he'd like 2 tasks for the promote. 1 for Test, and another for Prod. Child issues have been created for each.

**Comment by Sreelatha-BlazeKannan** [comment]
DEV to TEST Deployment plan sent on 07/17/2020

**Comment by TracyLinder** [comment]
Promote to Test complete.

**Comment by TracyLinder** [comment]
**Status:** Need to test this w/ Sangeeta. If she approves, then move to prod. Updating milestone to Sprint 30.

**Comment by TracyLinder** [comment]
**Update:**

Per Sree, some updates needed to be made and re-commited to test. We have scheduled a demo & UAT on 8/3: #3543

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader  - Testing is in progress, Completed testing for TIA, C8, ZC & SP folders. C testing is in progress.

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader  - Testing completed in TEST Environment and all folders are working as expected.

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan @TracyLinder I was able to test this on TEST and it met all of the acceptance criteria below:

'Transportation Mitigation Fee-in-lieu' is available on TIA, Insert & Run fee procedure
ATD Engineering Review process does not save an 'Approved' attempt result if there are unpaid ATD fees on the fee tab. This applies to folders SP, C8, ZC, and C.
ATD Engineering Review process provides attempt result 'Insert Mitigation Fee'
The Insert Mitigation Fee inserts the Transportation Mitigation Fee-in-lieu in the amount specified by the case manager on the process info.

**Comment by TracyLinder** [comment]
**Status:** In Test, pending customer UAT

Sangeeta has requested some time to do independant testing before this is approved for PROD. She'll be doing that early next week.

**Comment by TracyLinder** [comment]
**Email to Stakeholder:**

Hi Sangeeta,

I hope you are having a good week. I'm reaching out to see how testing is going?

**Comment by TracyLinder** [comment]
### Email Conversation w/ Sageeta:

**Sangeeta:**

Hi Tracy,
 
The testing worked well. One thing I noticed that in the ‘insert run’ menu I did not see the fee by name.  I entered it in the ‘insert’ by fee code, and it worked.  

**Tracy:**

Hi Sangeeta,

Which fee are you referring to that's missing from the Insert + Run? The mitigation fee?


**Sangeeta:**

Yes, the mitigation fee.

**Tracy:**

Thanks for the clarification. The fee was not programmed on ‘Insert + Run’ for two reasons:
1.	The system won’t allow this to insert with a $0 amount, so a dummy amount would have to be coded.
2.	To remind the user that the fee should be added using the new folder process features. 

Please let me know if this is acceptable, or if you’d like us to add the mitigation fee to Insert + Run with a dummy fee (like $100). 

**Sangeeta:**
Thanks Tracy.  Your explanation makes perfect sense.  It can go LIVE!

**Comment by TracyLinder** [comment]
### Approved to promote to Prod!! 🎉🎉

@Sreelatha-BlazeKannan

**Comment by Sreelatha-BlazeKannan** [comment]
@TracyLinder , @Nadin-Nader - PROD Deployment request submitted on 08/17/2020

**Comment by Nadin-Nader** [comment]
@Sreelatha-BlazeKannan  @TracyLinder  The DevOps Deployment TASK is SCTASK0090582. The Deployment for the LookUp is going to CTM Tango as it is a LookUp on PROD and i cant add it to PROD. I have submitted the task for this and it is SCTASK0090583.

**Status:**
Waiting on CTM  Tango and CM DevOps team to deploy to PROD

**Comment by Nadin-Nader** [comment]
@TracyLinder @Sreelatha-BlazeKannan Just talked to DEB in the stand up and she informed me that for some reason the on call that is supposed to be doing Deployments, didnt get a chance to.. she said Tuesday if not to message her.

**Comment by Nadin-Nader** [comment]
**STATUS:**
Robert will be deploying to PROD on Thursday 9/3/2020

**Comment by TracyLinder** [comment]
### This is in PROD ✨🎉
- SCTASK0090582 was promoted on 8/18
- SCTASK0090583 was promoted on 9/3

I will inform Sangeeta; I'm also going to schedule time with her to do a smoke test of these features in PROD. Closing this ticket.
